assertsprocessor/metrics.go

```diff
@@ -41,7 +41,8 @@ type metricHelper struct {
 	totalTraceCount    *prometheus.CounterVec
 	sampledTraceCount  *prometheus.CounterVec
 	// limit cardinality of request contexts for which metrics are captured
-	requestContextsByService *xsync.MapOf[string, *ttlcache.Cache[string, string]]
+	requestContextsByService *xsync.MapOf[string, *ttlcache.Cache[string, prometheus.Labels]]
```

Comment:
For a given request context, what is the likelihood that there are multiple labels? For e.g. both `Client` and `Server` span? Also, if we decide to include the response status code in the label, we may have multiple labels. So I think, it might be better to use the delete partial method and delete all metrics which have given value for the `namespace, service and request context` labels.